lab_2_tokenize_by_bpe/main.py

```diff
@@ -47,6 +80,23 @@ def merge_tokens(
     :param pair: a pair of tokens to be merged
     :return: dictionary in the form of <preprocessed word: number of occurrences>
     """
+    if (not isinstance(word_frequencies, dict) or
+        not isinstance(pair, tuple)):
+        return None
+    freq_2 = {}
+    for word in word_frequencies:
+        if ''.join(pair) in ''.join(word):
+            pair_i = []
+            word_list = list(word)
+            for i in range(len(word) - 1):
+                if (word[i], word[i + 1]) == pair:
+                    pair_i.append(i)
+            for i in pair_i:
+                word_list[i:i + 2] = [''.join(pair)]
```

Comment:
Congratulations on solving the secret! But there are some mistakes. Think about this line. It is in this place that you have problems and you get "ве" instead of "ле". Maybe debugging can be useful for you! Good luck!